src/test/java/mate/academy/security/AuthenticationServiceImplTest.java

```diff
@@ -0,0 +1,72 @@
+package mate.academy.security;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.RoleService;
+import mate.academy.service.UserService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class AuthenticationServiceImplTest {
+    private static final String ROLE_USER_NAME = "USER";
+    private static final String USER_EMAIL = "test@com.ua";
+    private static final String USER_PASSWORD = "password";
+    private UserService userService;
+    private RoleService roleService;
+    private AuthenticationService authenticationService;
+    private Role userRole;
+    private PasswordEncoder passwordEncoder;
+    private User testUser;
+
+    @BeforeEach
+    void setUp() {
```

Comment:
Iа objects are not modified or used differently in each test method, it can be considered a good practice to move them to a @BeforeAll method.
(passwordEncoder, authenticationService, etc.)
It's related to other tests as well.
And since these are tests, you can do them final and initialize as mock when declare